### PR TITLE
Use timeouts instead of relying on runtime.GoSched

### PR DESCRIPTION
**Pull request by cpuguy83**
ping @LK4D4 @aaronlehmann

### CODE REVIEW

File: pkg/locker/locker_test.go [hunk 1]
```diff
@@ -34,7 +34,22 @@ func TestLockerLock(t *testing.T) {
 		close(chDone)
 	}()
 
-	runtime.Gosched()
+	chWaiting := make(chan struct{})
+	go func() {
+		c := time.NewTicker(1 * time.Millisecond)
+		for range c.C {
```
Comment (LK4D4) [inline pkg/locker/locker_test.go]: looks like can be just `for range time.Tick(...)` :)

**Comment by aaronlehmann** [comment]
LGTM

**Comment by cpuguy83** [comment]
@LK4D4 Updated... hoped nobody would notice that :)

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**